src/plugins/layout/shadowdom.ts

```diff
@@ -365,8 +367,11 @@ export class ShadowDom {
     let newNodes = this.doc.getElementsByClassName(NewNodeClassName);
     for (let i = 0; i < newNodes.length; i++) {
       let shadowNode = newNodes[i] as IShadowDomNode;
+      let currentIndex = getNodeIndex(shadowNode);
       shadowNode.id = "" + nextIndex;
       shadowNode.node[NodeIndex] = nextIndex;
+      delete this.nodes[currentIndex];
```

Comment:
i think since nodes is just a lookup table we will be ok even long term. 

---
In reply to: [133095363](https://github.com/Microsoft/clarity-js/pull/27#discussion_r133095363) [](ancestors = 133095363)